File: lab2/task1/PRNG.cpp
#include <random>
#include <string>
#include <fstream>

void generate_random_sequence(const std::string &path)
{
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> dis(0, 1);
    std::string seq;
    for (int i = 0; i < 128; ++i)
        seq += std::to_string(dis(gen));
    std::ofstream file(path);
    file << seq;
}

int main()
{
    generate_random_sequence("seq_generator_cpp.txt");
}

File: lab2/task1/PRNG.java
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

public class PRNG {
    public static void generateRandomSequence(String path) {
        Random random = new Random();
        StringBuilder seq = new StringBuilder(128);

        for (int i = 0; i < 128; i++) {
            seq.append(random.nextInt(2));
        }

        try (FileWriter fileWriter = new FileWriter(path)) {
            fileWriter.write(seq.toString());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    public static void main(String[] args) {
        generateRandomSequence("seq_generator_java.txt");
    }
}